src/customerio-inapp.ts

```diff
@@ -51,35 +55,45 @@ class CustomerIOInAppMessaging implements NativeSpec {
     };
 
     // Holds the real subscription once created
-    let actualSubscription: EventSubscription | EmitterSubscription;
+    let actualSubscription: EmitterSubscription;
     let removed = false;
 
-    // Proxy object returned immediately to support .remove()
+    // Proxy object returned immediately to keep the API consistent
     const proxySubscription = {
       remove: () => {
         removed = true;
         actualSubscription?.remove();
       },
-    };
+      emitter: new NativeEventEmitter(),
+      eventType: InAppEventListenerEventName,
+      listener: emitter,
+      context: undefined,
+      key: 0,
+      subscriber: {} as any,
+    } as EmitterSubscription;
```

Comment:
Because new arch returns `EventSubscription` which is base class of `EmitterSubscription`, we have to fake other values. Should not impact customers as adding and removing works fine with it.